src/runtime-rs/crates/hypervisor/src/device/topology.rs

```diff
@@ -0,0 +1,293 @@
+//
+// Copyright (c) 2019-2023 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/*
+The design Origins from https://github.com/qemu/qemu/blob/master/docs/pcie.txt
+
+The flatten PCIe topology. Root Bus (pcie.0)
+pcie.0 bus (Root Complex)
+----------------------------------------------------------------------------
```

Comment:
this graph is kinda different from `https://github.com/qemu/qemu/blob/master/docs/pcie.txt`
Does this mean we are not gonna support device like `PCIe Root Port `, `PCIe-PCI Bridge` or `pxb-pcie`

Reply:
Comments added to explain it